src/Datadog.Trace/Agent/Api.cs

```diff
@@ -106,6 +106,7 @@ public async Task<bool> SendTracesAsync(Span[][] traces)
                     {
                         // stop retrying
                         Log.Error(exception, "An error occurred while sending traces to the agent at {0}", _tracesEndpoint);
+                        _statsd?.Send();
```

Comment:
I believe you should either remove this or rename the PR. You can't change the behavior of the tracer (even to fix a bug) in a PR names "unit test improvements"

Reply:
Yes good point, I'll move the bug fix to a separate PR 👍